java/memory/src/main/java/org/apache/arrow/memory/BufferAllocator.java

```diff
@@ -50,6 +50,27 @@
    */
   public ArrowBuf buffer(int size, BufferManager manager);
 
+  /**
+   * Allocate a new or reused buffer of the provided size. Note that the buffer may technically
+   * be larger or smaller than the  requested size, depending on the rounding option.
+   * However, the buffer's capacity will be set to the configured size.
+   *
+   * @param size The size in bytes.
+   * @param manager A buffer manager to manage reallocation.
+   * @param roundingOption The rounding option.
+   * @return a new ArrowBuf, or null if the request can't be satisfied
+   * @throws OutOfMemoryException if buffer cannot be allocated
+   */
+  public ArrowBuf buffer(int size, BufferManager manager, BaseAllocator.AllocationRoundingOption roundingOption);
+
+  /**
+   * Get the rounded buffer size.
+   * @param size The requested buffer size.
+   * @param roundingOption The rounding option.
+   * @return the buffer size after rounding.
+   */
+  public int getRoundedSize(final int size, BaseAllocator.AllocationRoundingOption roundingOption);
```

Comment:
does this need to be in the interface?

Reply:
@emkornfield Good question.
In this PR, we provide two solutions to the problem of OOM when there is sufficient memory:
1. By providing a rounding option when allocating Arrow buffer.
2. By allowing the users to estimate the buffer size, and adjust the buffer size accordingly to avoid OOM.

The two new methods in this interface corresponds to the two solutions above. So I think it may be better to provide this method in the interface.